src/iperf_tcp.c

```diff
@@ -454,6 +454,8 @@ iperf_tcp_connect(struct iperf_test *test)
 	printf("SNDBUF is %u, expecting %u\n", sndbuf_actual, test->settings->socket_bufsize);
     }
     if (test->settings->socket_bufsize && test->settings->socket_bufsize > sndbuf_actual) {
+    close(s);
+	freeaddrinfo(server_res);
```

Comment:
Thanks for the PR! This looks pretty good. Formatting nitpick: Indent line 457 to line up with lines 458-460. Also these two lines should probably also be added to the the error-handling around line 477.